app/controllers/api/stages_controller.rb

```diff
@@ -18,10 +16,19 @@ def clone
     end
   end
 
+  def duplicable
+    render json: @project.duplicable_stage
+  end
+
+  def put_duplicable
+    duplicable_stage.duplicable!
+    head :no_content
+  end
+
   private
 
   def stage_to_clone
-    Stage.find_by_param!(params.require(:stage_id))
+    Stage.find_by_id(params.require(:stage_id))
```

Comment:
I thought about that, but permalinks are not RESTful in my opinion. 

The permalink is a nicety, like porcelain, it is essentially an extension of the UI.